### ISSUE TITLE
Limit unack message

### ISSUE THREAD

**Issue by jonathanroze** [vcabbage/amqp#171]
Hi,

How can I limit my receiver to receive only 5 or another number of unack message to avoid to receive all messages that are in the queue ?

Thanks

**Comment by devigned** [comment]
I believe if you set `LinkCredit` as show in [this example](https://github.com/vcabbage/amqp/blob/e6aeb3d8e5ac68426935114ac2a430c32391e5a7/example_test.go#L57), it will limit your receiver from receiving more than the limit before providing a disposition for the message.

Also, there's pretty solid docs associated with that option here: https://godoc.org/pack.ag/amqp#LinkCredit.

**Comment by jonathanroze** [comment]
Thanks for your time Devigned, i already tried this property and my receive all messages that are in the queue !

**Comment by jonathanroze** [comment]
I tried to put sleep into my onMessage function and credit set to 10, but in this case, i can read just one by one message !

**Comment by devigned** [comment]
Share some code. Happy to give it a look.

**Comment by vcabbage** [comment]
I agree, without the code it's difficult to advise on what might be happening.

`onMessage` isn't part of this library's API, but you'd need to be handling them in separate goroutines to get more than one at a time.

For what you're asking for to work you also need to be negotiating `amqp.LinkSenderSettle(amqp.ModeUnsettled)` otherwise the broker could be sending you messages that are already settled and there's nothing to ack.
